File: MethodsAsync/callback.js
function soma(x, callback) {
    setTimeout(() => {
        return callback(null, x + 5000)
    }, 3000)
}

function resolveSoma(err, resultado) {
    if (err) throw err
    console.log(`Resultado com callback: ${resultado}`)
}

soma(3000, resolveSoma)

File: MethodsAsync/callback.lua
-- Definindo a função de callback
local function meuCallback(resultado)
    print("O resultado é:", resultado)
end

-- Função que recebe o callback
local function minhaFuncao(callback)
    -- Simulando uma operação assíncrona
    local resultado = 10
    -- Invocando o callback com o resultado
    callback(resultado)
end

-- Chamando a função que recebe o callback
minhaFuncao(meuCallback)
